File: greenorganizer/src/app/calculate/result/ResultComponent.java
export class ResultComponent implements OnInit, OnDestroy {
  closetLength: number = 0;
  private subscription: Subscription;

  constructor(private clothingService: ClothingService) {}

  ngOnInit() {
    this.subscription = this.clothingService.closetLength$.subscribe(length => {
      this.closetLength = length;
    });
  }

  ngOnDestroy() {
    this.subscription.unsubscribe();
  }
}

File: greenorganizer/src/app/calculate/result/result.component.ts
import {Component, Input, OnInit} from '@angular/core';
import {ClothingService} from "../../clothing.service";


@Component({
  selector: 'app-result',
  templateUrl: './result.component.html',
  styleUrls: ['./result.component.css']
})
export class ResultComponent implements OnInit {
  @Input() closetLength: number = 0;

  constructor(private clothingService: ClothingService) {}

  ngOnInit() {
    this.closetLength = this.clothingService.calculateSustainabilityScore();
  }

}
